sql-plugin/src/main/spark330/scala/com/nvidia/spark/rapids/GpuBloomFilterMightContain.scala

```diff
@@ -0,0 +1,104 @@
+import com.nvidia.spark.rapids.shims.ShimBinaryExpression
+
+import org.apache.spark.TaskContext
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.expressions.{Expression, GetStructField, PlanExpression}
+import org.apache.spark.sql.catalyst.trees.TreePattern.OUTER_REFERENCE
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+case class GpuBloomFilterMightContain(
+    bloomFilterExpression: Expression,
+    valueExpression: Expression)
+  extends ShimBinaryExpression with GpuExpression with AutoCloseable {
+
+  @transient private lazy val bloomFilter: GpuBloomFilter = {
+    Option(TaskContext.get).foreach(_.addTaskCompletionListener[Unit](_ => close()))
+    withResourceIfAllowed(bloomFilterExpression.columnarEvalAny(new ColumnarBatch(Array.empty))) {
+      case s: GpuScalar => GpuBloomFilter(s)
+      case x => throw new IllegalStateException(s"Expected GPU scalar, found $x")
+    }
+  }
+
+  override def nullable: Boolean = true
+
+  override def left: Expression = bloomFilterExpression
+
+  override def right: Expression = valueExpression
+
+  override def prettyName: String = "might_contain"
+
+  override def dataType: DataType = BooleanType
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    (left.dataType, right.dataType) match {
+      case (BinaryType, NullType) | (NullType, LongType) | (NullType, NullType) |
+           (BinaryType, LongType) =>
+        bloomFilterExpression match {
+          case e: Expression if e.foldable => TypeCheckResult.TypeCheckSuccess
+          case subquery: PlanExpression[_] if !subquery.containsPattern(OUTER_REFERENCE) =>
+            TypeCheckResult.TypeCheckSuccess
+          case GetStructField(subquery: PlanExpression[_], _, _)
+            if !subquery.containsPattern(OUTER_REFERENCE) =>
+            TypeCheckResult.TypeCheckSuccess
+          case _ =>
+            TypeCheckResult.TypeCheckFailure(s"The Bloom filter binary input to $prettyName " +
+              "should be either a constant value or a scalar subquery expression")
+        }
+      case _ => TypeCheckResult.TypeCheckFailure(s"Input to function $prettyName should have " +
+        s"been ${BinaryType.simpleString} followed by a value with ${LongType.simpleString}, " +
+        s"but it's [${left.dataType.catalogString}, ${right.dataType.catalogString}].")
+    }
+  }
+
+  override def columnarEval(batch: ColumnarBatch): GpuColumnVector = {
+    if (bloomFilter == null) {
+      GpuColumnVector.fromNull(batch.numRows(), dataType)
+    } else {
+      withResource(valueExpression.columnarEval(batch)) { value =>
+        if (value == null || value.dataType == NullType) {
+          GpuColumnVector.fromNull(batch.numRows(), dataType)
+        } else {
+          GpuColumnVector.from(bloomFilter.mightContainLong(value.getBase), BooleanType)
+        }
+      }
+    }
+  }
+
+  override def close(): Unit = {
```

Comment:
When is this called?

Reply:
When `bloomFilter` is lazily evaluated here: https://github.com/NVIDIA/spark-rapids/pull/8775/files#diff-95125ce9b0fef86d71d7b20aa1de5d59ab49239f26d13536961e436a6d74e44cR47